### ISSUE TITLE
Name clash with an old Java IDE

### ISSUE THREAD

**Issue by hcarty** [janestreet/jbuilder#188]
Maybe no longer relevant - https://en.wikipedia.org/wiki/JBuilder - but worth noting!  Pointed out by a colleague.

**Comment by diml** [comment]
Yh, I know the name is rubbish... I'm just not good at finding names

**Comment by rdavison** [comment]
Are you inviting us to bike shed the name? :)

**Comment by diml** [comment]
Haha, sure go on

**Comment by hcarty** [comment]
Oh no!  My actions have had unforeseen (but in hindsight easily foreseeable...) consequences!

**Comment by lpw25** [comment]
I think it would be best to just switch the name to `obuilder`. A quick google doesn't show any real conflicts, and it is close enough to `jbuilder` that it shouldn't cause much confusion. I suppose people might confuse it with `obuild`, but there are only so many combinations of `OCaml` and `build` that you can make. The alternative would be to just pick some random name, but that seems like it would lead to a lot of bike-shedding.

Updating the documentation, adding the packages to opam and an announcement on the discuss forum, the caml-list and maybe /r/ocaml should be all that's needed to make the switch.

**Comment by lpw25** [comment]
Also, this really needs to happen before the `1.0` release if possible.

**Comment by dra27** [comment]
It must be dealt with before the 1.0 release - CodeGear still have a valid trademark on it (https://trademarks.ipo.gov.uk/ipo-tmcase/page/Results/4/EU002533685)

**Comment by dra27** [comment]
@lpw25's suggestion of `obuilder` sounds good to me (when said aloud, `obuilder build` has a suitably deific ring to it, as an invocation!).

Given that there's no risk of conflict with the actual software, we could continue to offer a jbuilder command as well, just with a deprecation warning.

**Comment by yminsky** [comment]
Does that mean we should change from jbuild files to obuild files?

**Comment by dra27** [comment]
Prefer `obuild` and fall back to `jbuild` otherwise, again with a deprecation warning?

**Comment by rgrinberg** [comment]
IMO name clashes such as this are unavoidable and what's important is that jbuilder is a unique name among ocaml software. Changing the name will cause nothing but pointless churn.

**Comment by dra27** [comment]
@rgrinberg - I'm afraid your opinion is in conflict with law!

**Comment by samoht** [comment]
can we rename `jbuilder` to `opam-build`? This will allow people to write:

```
opam build
```

and the project will just build :-) Which also means that you don't need to specify `build` and `buildtest` commands anymore, by default they will just use the current `jbuilder` runes that everybody is using.

**Comment by diml** [comment]
In the end we do need to change the name, I created this ticket for gathering suggestions: https://github.com/janestreet/jbuilder/issues/360

**Comment by rgrinberg** [comment]
Do we also need a ticket to detail the work required for a painless
transition

**Comment by diml** [comment]
Sounds good

**Comment by rgrinberg** [comment]
I'm closing this ticket because we already have a ticket for brainstorming suggestions and a ticket to do the actual rename.
